python/caffe/_caffe.cpp

```diff
@@ -115,10 +115,18 @@ void Net_Save(const Net<Dtype>& net, string filename) {
 }
 
 void Net_SetInputArrays(Net<Dtype>* net, bp::object data_obj,
-    bp::object labels_obj) {
+    bp::object labels_obj, const string &layer_name) {
+
+  shared_ptr<MemoryDataLayer<Dtype> > md_layer;
+  if (layer_name == "") {
```

Comment:
None in Python is `bp::object()` in C++, if that helps this comparison here. You can use `bp::extract<std::string>(layer_name)` to get the string out as well.